62.CAD/Polyline.h

```diff
@@ -255,10 +281,223 @@ class CPolyline
+		{
+			// TODO: we already calculate quadratic form of each bezier (except of the last one), maybe store this info in an array and use it later to calculate normals?
+			const float64_t2 bezierDerivativeValueAtP0 = 2.0 * (quadBezierInfo.p[1] - quadBezierInfo.p[0]);
+			const float32_t2 tangentAtP0 = glm::normalize(bezierDerivativeValueAtP0);
+			//const float_t2 A = P0 - 2.0 * P1 + P2;
+			const float64_t2 bezierDerivativeValueAtP2 = 2.0 * (quadBezierInfo.p[0] - 2.0 * quadBezierInfo.p[1] + quadBezierInfo.p[2]) + 2.0 * (quadBezierInfo.p[1] - quadBezierInfo.p[0]);
+			const float32_t2 tangentAtP2 = glm::normalize(bezierDerivativeValueAtP2);
+
+			PolylineConnectorNormalHelperInfo connectorNormalInfoAtP0{};
+			connectorNormalInfoAtP0.worldSpaceCircleCenter = quadBezierInfo.p[0];
+			connectorNormalInfoAtP0.normal = float32_t2(-tangentAtP0.y, tangentAtP0.x);
+			connectorNormalInfoAtP0.phaseShift = phaseShift;
+			connectorNormalInfoAtP0.type = ObjectType::QUAD_BEZIER;
+
+			PolylineConnectorNormalHelperInfo connectorNormalInfoAtP2{};
+			connectorNormalInfoAtP2.worldSpaceCircleCenter = quadBezierInfo.p[2];
+			connectorNormalInfoAtP2.normal = float32_t2(-tangentAtP2.y, tangentAtP2.x);
+			connectorNormalInfoAtP2.phaseShift = phaseShift;
+			connectorNormalInfoAtP2.type = ObjectType::QUAD_BEZIER;
+
+			connectorNormalInfos.push_back(connectorNormalInfoAtP0);
+			connectorNormalInfos.push_back(connectorNormalInfoAtP2);
+		}
+
+		std::vector<PolylineConnector> buildConnectors()
+		{
+			std::vector<PolylineConnector> connectors;
+
+			if (connectorNormalInfos.size() < 2u)
+				return {};
+
+			uint32_t i = getConnectorNormalInfoCountOfLineType(connectorNormalInfos[1].type);
+			while (i < connectorNormalInfos.size())
+			{
+				const auto& prevLine = connectorNormalInfos[i - 1];
+				const auto& nextLine = connectorNormalInfos[i];
+
+				const float32_t2 prevLineNormal = connectorNormalInfos[i - 1].normal;
+				const float32_t2 nextLineNormal = connectorNormalInfos[i].normal;
+
+				const float crossProductZ = nbl::hlsl::cross2D(nextLineNormal, prevLineNormal);
+				constexpr float EPSILON = 0.000001f;
+				const bool isMiterVisible = std::abs(crossProductZ) >= EPSILON;
+				if (isMiterVisible)
+				{
+					const float64_t2 intersectionDirection = glm::normalize(prevLineNormal + nextLineNormal);
+					const float64_t cosAngleBetweenNormals = glm::dot(prevLineNormal, nextLineNormal);
+
+					PolylineConnector res{};
+					res.circleCenter = connectorNormalInfos[i].worldSpaceCircleCenter;
+					res.v = static_cast<float32_t2>(intersectionDirection * std::sqrt(2.0 / (1.0 + cosAngleBetweenNormals)));
+					res.cosAngleDifferenceHalf = static_cast<float32_t>(std::sqrt((1.0 + cosAngleBetweenNormals) * 0.5));
+					res.phaseShift = connectorNormalInfos[i].phaseShift;
+
+					const bool needToFlipDirection = crossProductZ < 0.0f;
+					if (needToFlipDirection)
+					{
+						res.v = -res.v;
+					}
+					res.v.y = -res.v.y;
```

Comment:
comment why you do this here. // Negating y to avoid doing it in vertex shader when working in screen space where y is in the opposite direction of worldspace y direction